java-checks/src/main/java/org/sonar/java/checks/FieldModifierCheck.java

```diff
@@ -31,10 +32,12 @@
 import org.sonar.plugins.java.api.tree.VariableTree;
 
 import java.util.List;
+import java.util.function.Predicate;
 
 @Rule(key = "S2039")
 public class FieldModifierCheck extends IssuableSubscriptionVisitor {
 
+  private static final Predicate<ModifiersTree> MODIFIERS_TREE_PREDICATE = new VisibleForTestingPredicate().negate();
```

Comment:
There are integration test just because they use multiple classes. In fact the whole java parser/lexer is used. That's not unit testing.

I don't mind you like integration tests better then unit testing. In fact many companies/developers like that. It's most effective to reach 80% coverage  in that way,  bit extreamly difficult for 100% of conditions. Unit testing let you reach 100% quite easily because you focus on every edge case in single class so you know that class is surly working. But still you need integration test to verify yourself between multiple classes. 

I mind copy pasting of my code. Even if it's 10 lines. I'm pretty much sure sonar could detect that as copy paste. :-/